onnx/checker.cc

```diff
@@ -35,15 +43,138 @@ namespace checker {
     }                                         \
   } while (0)
 
-void check_value_info(const ValueInfoProto& value_info, const CheckerContext&) {
+bool check_name_syntax(const std::string& name) {
+  if (name.empty()) {
+    return false;
+  }
+
+  bool correct = true;
+
+  // Names should adhere to C identifier syntax.
+  auto iter = name.cbegin();
+
+  char c = *iter;
+  if (!(isalpha(c) || c == '_')) {
+    correct = false;
+  }
+
+  ++iter;
+
+  for (; iter < name.cend(); ++iter) {
+    c = *iter;
+    if (!(isalnum(c) || c == '_')) {
+      correct = false;
+    }
+  }
+
+  return correct;
+}
+
+bool check_domain_syntax(const std::string& name) {
+  if (name.empty()) {
+    return false;
+  }
+
+  bool correct = true;
+
+  // Names should adhere to C identifier syntax.
```

Comment:
'.' (dot) is not in C identifier syntax

Reply:
Thanks for catching the inconcistency. I will update the comment in the function to reflect that this is about domain name syntax, which does allow '.'